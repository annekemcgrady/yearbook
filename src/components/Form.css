.form {
  height: 50px;
  
}

input, button {
  height: 40px;
  width: 200px;
  border: 1px solid magenta;
  margin: 10px;
  font-size: 1em;
}

button {
  background-color: limegreen;
  color: magenta;
}


