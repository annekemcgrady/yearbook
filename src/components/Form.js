import React, { Component } from 'react';
import './Form.css';

//yearbook add new student form

class Form extends Component {
  constructor() {
    super()
    this.state = {
      name: '',
      quote: '',
      superlative: ''
    }
  }

  handleChange = (event) => {
    this.setState({ [event.target.name] : event.target.value })
  }

  submitNewStudent = (event) => {
    event.preventDefault()
    const newStudent = {...this.state, id: Date.now()};
    this.props.addNewStudent(newStudent)
    this.resetInputs()
  }

  resetInputs = () => {
    this.setState( {
      name: '',
      quote: '',
      superlative: ''
    })
  }

  render() {
    return (
    <form className='form'>
      <input 
        name='name' 
        type='text'
        placeholder='Enter name'
        value={this.state.name}
        onChange={this.handleChange}
        />
        <input 
        name='quote' 
        type='text'
        placeholder='Enter favorite quote'
        value={this.state.quote}
        onChange={this.handleChange}
        />
        <input 
        name='superlative' 
        type='text'
        placeholder='Enter favorite superlative'
        value={this.state.superlative}
        onChange={this.handleChange}
        /> 
        <button onClick={this.submitNewStudent}>Create Student</button>
    </form>
    )
  }
}

export default Form;