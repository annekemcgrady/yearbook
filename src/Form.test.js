import React from 'react';
import { shallow, mount } from 'enzyme';
import Form from './components/Form.js';

describe('Form', ()=> {
  it('should match the snapshot', ()=> {
    const mockMethod = jest.fn()
    const wrapper = mount(
      <Form 
        addNewStudent={mockMethod}
      />);

      expect(wrapper).toMatchSnapshot();
  })

  it('should match the snapshot', ()=> {
    const mockMethod = jest.fn()
    const wrapper = mount(
      <Form 
        addNewStudent={mockMethod}
      />);

    wrapper.find('.new-submit').simulate('click');
    expect(mockMethod).toBeCalled(); 
    })
})

