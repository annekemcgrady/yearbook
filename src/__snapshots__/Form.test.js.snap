// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should match the snapshot 1`] = `
<Form
  addNewStudent={[MockFunction]}
>
  <form
    className="form"
  >
    <input
      name="name"
      onChange={[Function]}
      placeholder="Enter name"
      type="text"
      value=""
    />
    <input
      name="quote"
      onChange={[Function]}
      placeholder="Enter favorite quote"
      type="text"
      value=""
    />
    <input
      name="superlative"
      onChange={[Function]}
      placeholder="Enter favorite superlative"
      type="text"
      value=""
    />
    <button
      className="new-submit"
      onClick={[Function]}
    >
      Create Student
    </button>
  </form>
</Form>
`;
